content/articles/online-machine-learning-with-river-python/index.md

```diff
@@ -0,0 +1,446 @@
+
+#### Looping through our dataset
+
+```python
+for text,label in data:
+    pipe_nb = pipe_nb.learn_one(text,label)
+```
+
+When looping through our dataset we use the `learn_one()` method to learn one at a time from our given list of the tuple. `learn_one()` will learn from the first `text` and `label` which is `"my python program is running", "software"` according to our data set.
+
+It will then store this knowledge learned and use it when the next data arrives. Since the model remembers the knowledge gained over time, it will be able to adapt to changes in the dataset. Over time, the model will be more accurate than the beginning of the training since it would have accumulated knowledge.
+
+### Making prediction
+
+Since this is online machine learning, the model does not wait till the end to make predictions. It will predict this instance and continue training when the next data arrives.
+
+```python
+pipe_nb.predict_one("I built an API")
+```
+
+We use `predict_one` to predict according to this instance.
+
+The output is shown:
+
+```bash
+'software'
+```
+
+This is true since our text is `software` related.
+
+### Prediction probability
+
+To get the probability of the above classification use this command.
+
+```python
+pipe_nb.predict_proba_one("I built an API")
+```
+
+We use the `predict_proba_one()` method to get the probability of prediction at this instance.
+The output is as shown.
+
+```python
+{'software': 0.732646964375691, 'hardware': 0.2673530356243093}
+```
+
+This shows that the probability of the text being classified as `software` is more than that of `hardware`. Their probabilities are `0.7326` and `0.2673` respectively.
+
+Online machine learning is a continuous process since we are real-time time data that is continuously being generated.
+
+Let's make another prediction.
+
+#### Other prediction
+
+```python
+pipe_nb.predict_one("the hard drive  in the computer is damaged")
+```
+
+The prediction output:
+
+```bash
+'software'
+```
+
+At the beginning of the training phase, the model might give a lower accuracy but with time the accuracy increases.
+This is because the model stores the knowledge it has learned at each point making it more accurate.
+
+### Model accuracy
+
+We need to get the accuracy of our model at this instance. We use the `river.metrics.Accuracy()` method to calculate the accuracy.
+
+We loop through our data set to get the instance we want to calculate the accuracy and update our model with the accuracy score after making a prediction using `metric.update`.
+
+```python
+metric = river.metrics.Accuracy()
+for text,label in test_data:
+    y_pred_before = pipe_nb.predict_one(text)
+    metric = metric.update(label,y_pred_before)
+    pipe_nb = pipe_nb.learn_one(text,label)
+```
+
+To get the accuracy score use the following command.
+
+```python
+metric
+```
+
+The output is as shown.
+
+```bash
+Accuracy: 75.00%
+```
```

Comment:
Please explain in a few words what these percentage denotes. What are the significance of this result? Explain to the reader